snafu/benchmarks/coremarkpro/coremarkpro.py

```diff
@@ -0,0 +1,207 @@
+            table_name = ""
+            results = {}
+            for line in file:
+                line = line.rstrip()
+                if not line:
+                    continue
+
+                # Find where the table starts and skips the fluff
+                if "RESULTS TABLE" in line:
+                    table_name = line.split(' ')[0].lower()
+                    while True:
+                        # Exit out of loop once it finds the table delimiter
+                        if('---' in next(file)):
+                            break
+                    continue
+
+                cols = re.split(r'\s+', line.rstrip())
+                converted_cols = [func(val) for func, val in zip(types, cols)]
+                record = dict(zip(headers, converted_cols))
+                record['type'] = table_name
+                test_results.append(record)
+
+        return test_results
+
+    def submit_results(self, sample_starttime) -> BenchmarkResult:
+
+        test_config = {
+                "workload": self.config.workload,
+                "context": self.config.context,
+                }
+        result_summary = {
+                #"clustername": self.config.cluster_name,
+                "date": sample_starttime,
+                "sample": self.config.sample,
+                "test_config": test_config,
+                }
+
+        test_results = {
+                "summary_results": self.parse_marks(),
+                "raw_results": self.parse_raw_logs()
+                }
+
+        return self.create_new_result(
+                data=test_results,
+                config=result_summary,
+                tag="results",
+                )
+
+    def convert_coremark_timestamp(self, timestamp):
+        from dateutil import tz
+        time_obj = datetime.strptime(timestamp, '%y%j:%H:%M:%S')
+        utc_tz = tz.gettz('UTC')
+
+        return (time_obj.astimezone(utc_tz)).strftime("%Y-%m-%dT%H:%M:%S.%fZ")   
+
+
+    """ Override member functions from Benchmark class"""
+
+    def setup(self) -> bool:
+
+        # Parse the command line args
+        self.config.parse_args()
+
+        # Sets up defaults for the required variables
+        self.config.uuid = os.getenv("uuid", str(uuid.uuid4()))
+        self.config.user = os.getenv("test_user", "myuser")
+        if "clustername" not in os.environ:
+            self.config.cluster_name = "mycluster"
+
+        return True
+
+    def collect(self) -> Iterable[BenchmarkResult]:
+
+        cmd = self.build_workload_cmd()
+
+        if not self.config.upload:
+            for sample_num in range(1, self.config.sample + 1):
+                self.logger.info(f"Starting coremark-pro sample number {sample_num}")
+
+                sample_starttime = datetime.utcnow().strftime("%Y-%m-%dT%H:%M:%S.%fZ")
+
+                # Runs the actual command
+                sample: ProcessSample = sample_process(
```

Comment:
This class runs the command and provides some added functionality. See https://github.com/cloud-bulldozer/benchmark-wrapper/blob/master/snafu/benchmarks/uperf/uperf.py for how to parse the output